src/config/config.rs

```diff
@@ -0,0 +1,229 @@
+use crate::common::error::DatenLordError;
+use clap::Parser;
+use std::net::IpAddr;
+use std::str::FromStr;
+
+use super::inner::InnerConfig;
+use super::inner::Role;
+use super::inner::StorageConfig as InnerStorageConfig;
+use super::inner::StorageParams as InnerStorageParams;
+use super::inner::StorageS3Config as InnerStorageS3Config;
+
+#[derive(Debug, Parser)]
+#[clap(author, version, about, long_about = None)]
+/// A config
+pub struct Config {
+    #[clap(long = "role", value_name = "VALUE")]
+    /// Role: Controller, Node, AsyncFuse, Scheduler
+    pub role: String,
+    #[clap(long = "node-name", value_name = "VALUE")]
+    /// Node name
+    pub node_name: String,
+    #[clap(long = "node-ip", value_name = "VALUE")]
+    /// Node ip
+    pub node_ip: String,
+    #[clap(long = "mount-path", value_name = "VALUE")]
+    /// Mount path
+    pub mount_path: String,
+    #[clap(flatten)]
+    /// Storage related config
+    pub storage: StorageConfig,
+}
+
+impl TryInto<InnerConfig> for Config {
```

Comment:
Shall we implement `TryFrom<Config>` for `InnerConfig`, as recommended by the official documentation? The `TryInto` trait will be implemented automatically.

Reply:
Ok , I will use `TryFrom`.